src/Source.h

```diff
@@ -47,6 +47,47 @@ class Source {
     return cur;
   }
 
+  static const char* skipBom(const char* begin) {
+    // Skip any unicode byte order marks
+    switch(begin[0]) {
+      // UTF-32BE
+      case '\x00':
+        if (begin[1] == '\x00' && begin[2] == '\xFE' && begin[3] == '\xFF') {
```

Comment:
Using an interator_range will likely have more overhead than doing the comparisons directly, the main benefit of the current code is each byte is compared only once, using any ranged compare would require you to compare the first few bytes more than once for each comparison.

I can make the current safe by passing the end ptr and verifying the length is at least (2 or 4) prior to comparison.